Add test configuration for 5-galaxy Exabox inter-mesh routing validation

jpanasiukTT opened the pull request:
### Ticket
https://github.com/tenstorrent/tt-metal/issues/32278

### Problem description

- No existing test configuration for 5-galaxy setup with comprehensive traffic patterns

### What's changed

- Added test suite with traffic patterns ordered by complexity:
  - Basic unicast (single sender → single receiver)
  - Single sender → multiple receivers
  - Multiple senders → single receiver
  - Training patterns (hub-based forward/backward pass)
  - Sequential all-to-all (hardest, deadlock-safe pattern)

### Testing

On one of the 5 all to all hosts:
`ssh local-asaigal@wh-glx-a03u02`
Sync envs:
`./multihost_dev.sh --sync --run`
5 Galaxy Reset:
`mpirun-ulfm --host wh-glx-a03u02,wh-glx-a03u08,wh-glx-a03u14,wh-glx-a04u08,wh-glx-a04u14 --mca btl self,tcp --mca hwloc_base_binding_policy none --tag-output tt-smi -glx_reset > mpi_output.log`
Run newly created tests
`tt-run --rank-binding tests/tt_metal/distributed/config/5_galaxy_wh_exabox_rank_bindings.yaml --mpi-args "--host wh-glx-a03u02,wh-glx-a03u08,wh-glx-a03u14,wh-glx-a04u08,wh-glx-a04u14 --map-by rankfile:file=rankfile.txt --allow-run-as-root --mca btl self,tcp --mca btl_tcp_if_include ens5f0np0 --tag-output"  ./build/test/tt_metal/perf_microbenchmark/routing/test_tt_fabric --test_config tests/tt_metal/tt_metal/perf_microbenchmark/routing/test_5_galaxy_multi_mesh.yaml`


Outcome:
Successfully completes the test and closes user mode device drivers, test can be run repeatedly.

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/docs/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] [Galaxy quick](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-quick.yaml) CI passes (if applicable)
- [ ] [Galaxy demo tests, for Llama](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-demo-tests.yaml) CI passes, if applicable, because of current Llama work
- [ ] (For runtime and ops writers) [T3000 unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml) CI passes (if applicable, since this is run on push to main)
- [ ] (For models and ops writers) [T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) CI passes (if applicable, since this is required for release)
- [ ] New/Existing tests provide coverage for changes

tt-asaigal committed:
5 all to all rank bindings

tt-asaigal committed:
Working traffic 5 all to all tests

jpanasiukTT commented:
Rank bindings and mgd2.0 were already added in https://github.com/tenstorrent/tt-metal/pull/31610 so this only adds traffic tests

tt-asaigal commented on tests/tt_metal/tt_metal/perf_microbenchmark/routing/test_5_galaxy_multi_mesh.yaml:
```diff
@@ -0,0 +1,204 @@
+  # ======================================================================================
+  # All-to-All Pattern (Sequential)
+  # ======================================================================================
+  - name: "FiveGalaxySequentialAllToAll"
```
Add a comment here - mentioning that this is a stress test and when ported to pipelines, this should be treated as such.

tt-asaigal committed:
make tests cleaner and more descriptive

jpanasiukTT commented:
/codeowners ping

tt-asaigal commented on tests/tt_metal/tt_metal/perf_microbenchmark/routing/test_5_galaxy_multi_mesh.yaml:
```diff
@@ -0,0 +1,138 @@
+
+    parametrization_params:
+      size: [64, 128]
+      num_packets: [50]
```
I think this file can be a combination of stress tests and sanity tests.

I see the `FiveGalaxySingleSenderMultiReceiver` and `FiveGalaxyMultiSenderSingleReceiver` tests as sanity.

This test simulates the traffic pattern that the training folks use in their 3 tier setup - so we can make this a stress test.

I would sweep over more packet sizes and run with more packets:
`
size: [128, 512, 1024, 2048, 4096]
num_packets: [250]
`

tt-asaigal committed:
Make FiveGalaxyHubAndSpoke a stress test with comprehensive parameter sweep

- Updated size parameter to sweep [128, 512, 1024, 2048, 4096]
- Updated num_packets to 250 for more thorough testing
- Reflects training workload simulation as per review feedback
